SAML Integrations (#807)
clerk/clerk-docs: docs/authentication/saml/okta.mdx

---
title: Okta Workforce
description: Learn how to integrate Okta Workforce with Clerk using SAML SSO.
---

# Okta Workforce

<TutorialHero
  beforeYouStart={[
    {
      title: "Set up a Clerk application",
      link: "/docs/quickstarts/setup-clerk",
    },
    {
      title: "Add the appropriate Clerk SDK to your application",
      link: "/docs/quickstarts/overview",
    },
  ]}
>

- Use Okta Workforce to enable single sign-on (SSO) via SAML for your Clerk application.

</TutorialHero>

> Integrating Okta Workforce with Clerk allows you to use it as an Identity Provider (IdP) for your Clerk application to authenticate your users and manage their access to your application via SAML SSO.

Reviewer: ```suggestion
You can use Okta Workforce as a SAML SSO Identity Provider (IdP) for user authentication in your Clerk apps.
```
Just trying to compress this sentence a bit, as it's a lot to take in

Reviewer 2: We basically have that in the "You will learn" section in the tutorial hero, so what do you think about removing the intro paragraph altogether?

Reviewer: That works too!

## Tutorial

<Steps>

### Set up Enterprise Connection in Clerk

To create a SAML connection in Clerk:

1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com).
2. In the navigation sidebar, select **User & Authentication > [Enterprise Connections](https://dashboard.clerk.com/last-active?path=user-authentication/enterprise-connections)**.
3. Select the **Create connection** button.
4. You will be presented with a modal to create a new connection. Fill in the required fields.

> - **Name** - A unique name for the connection. This is so you can tell your connections apart.

Reviewer: To nest this bulleted list, it needs to be double-indented. It'll look like this:
![image](https://github.com/clerk/clerk-docs/assets/5156915/34ad47a0-6d28-47ec-b350-a7d24f4401d5)

Instead of this:
![image](https://github.com/clerk/clerk-docs/assets/5156915/e444d515-f76a-40d7-a246-4afdc12a03c2)

Also, IMO the only item that should be here is the instructions for choosing Okta Workforce under Identity Provider. The other two fields should be understandable within the UX, and if not we should make suggestions to the dashboard team so the name and domain field's purposes are clearer

Reviewer 2: Okay I'll remove the bullet points and change up the instruction.

- **Domain** - The email domain for which you will enable Enterprise SSO.
- **Identity Provider** - The IdP you are using. For this tutorial, select **Okta Workforce**.

> 5. Once you have filled in these fields, you will be directed to the **Connection** page which has the configuration settings for the connection you just created. **Do not select the Enable connection toggle just yet.** You need to fully configure your IdP first before exposing your connection to your users while you're in the, sometimes long, process of configuring and testing it.

Reviewer: ```suggestion
5. Next, on the **Connection** page, **fully configure your IdP before selecting the Enable connection toggle**. Doing so will prevent your connection from being exposed to your users before you configure and test it.
```
Some language simplification

Reviewer 2: What do you think about:

```suggestion
5. Next, on the **Connection** page, **do not select the Enable connection toggle just yet.** You need to fully configure your IdP first before exposing your connection to your users.
```

Reviewer: Looks good!

6. Leave this page open. You will need to come back to it to complete the setup.

### Create a new Enterprise Application in Okta

To create a new Enterprise Application in Okta:

1. Navigate to [Okta](https://www.okta.com/) and sign in.
2. In the Okta dashboard, select **Admin** in the top right corner.
3. In the navigation sidebar,  select the **Applications** dropdown and select **Applications**.
4. Select the **Create App Integration** button.
5. In the **Create a new app integration** modal, select the **SAML 2.0** option and select the **Next** button.

> 6. Once redirected to the **Create SAML Integration** page, fill in the **General Settings** fields. An **App name** is required.  You can also upload an **App logo** if you wish.

Reviewer: ```suggestion
6. Once redirected to the **Create SAML Integration** page, fill in the **General Settings** fields. An **App name** is required.
```
Not a big deal, but might as well cut any instructions that aren't strictly necessary

7. Select the **Next** button to continue.

### Configure your service provider

Once you have moved forward from the **General Settings** instructions, you will be presented with the **Configure SAML** page.

> To configure your service provider (Clerk), you will need to add these two fields to your IdP's application:

Reviewer: ```suggestion
To configure your service provider (Clerk), you must add these two fields to your IdP's application:
```

> - **Single sign-on URL** - This is a unique identifier for your SAML connection that your IdP application needs.

Reviewer: Should we make any suggestions on how to choose this identifier? Any best practices?

> - **Audience URI (SP Entity ID)** - This is your application's URL that your IdP will redirect your users back to after they have authenticated in your IdP.

Reviewer: Same question as above

> ---

Reviewer 3: We should Okta Workforce as the IdP name in this page

> title: Integrate Okta SAML with Clerk
> description: Learn how to integrate .

Reviewer 3: ```suggestion
title: Okta Workforce
description: Learn how to integrate Okta Workforce via SAML SSO.
```

---

> # Integrate Okta SAML with Clerk

Reviewer 3: ```suggestion
# Okta Workforce
```

<TutorialHero
  beforeYouStart={[
    {
      title: "Set up a Clerk application",
      link: "/docs/quickstarts/setup-clerk",
    },
    {
      title: "Set up a local project",
      link: "/docs/quickstarts/overview",
    },
  ]}
>

> - Use Okta as an identity provider for your Clerk application to authenticate users and manage their access to your application.

Reviewer 3: ```suggestion
- Use Okta Workforce to enable single sign-on (SSO) via SAML for your Clerk application.
```

</TutorialHero>

> Integrating Okta with Clerk allows you to use Okta as an identity provider for your Clerk application. This means that you can use Okta to authenticate users and manage their access to your application.

Reviewer 3: ```suggestion
Integrating Okta Workforce with Clerk allows you to use it as an Identity Provider for your Clerk application to authenticate your users and manage their access to your application via SAML SSO.
```

## Tutorial

<Callout type="info">
  You must have a local project with Clerk set up already to follow this tutorial. See [the quickstart docs](/docs/quickstarts/overview) to get started.
</Callout>

<Steps>

> ### Setup Clerk SAML Connection

Reviewer 3: ```suggestion
### Setup Clerk Enterprise Connection
```

1. Create a new Enterprise Connection in the Clerk Dashboard. 
2. Fill out the form with the details for your Okta Application.

<Images 
  width={3024}
  height={1652}
  src="/docs/images/authentication/saml/okta/create-enterprise-connection-clerk-dash.avif"
  alt="Create a new Enterprise Connection in the Clerk Dashboard"
/>

### Find the Okta Enterprise Applications 

<Images 
  width={3024}
  height={1652}
  src="/docs/images/authentication/saml/okta/find-create-okta-application.avif"
  alt=""
/>

###  Create a new Okta Enterprise Application

<Images 
  width={3024}
  height={1652}
  src="/docs/images/authentication/saml/okta/choose-saml-in-app-create.avif"
  alt=""
/>

> ### Find Okta Browser Plugin

Reviewer 3: Is this step actually needed?

Author: I can remove it :)
